### PR TITLE
Task 7 Post comment by article Id

### PR DESCRIPTION
**Pull request by WanwanShi**
Added success post comment path, invalid article_id type test/ out-of-range article_id test/ missing required request filed test, empty body post tests.

Added the corresponding functions

Added to the endpoints.json .

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -188,4 +188,87 @@ describe("/api/articles/:article_id/comments", ()=> {
...
+            body:"This is just something I want to test, Thank you!"
+        })
+        .expect(201)
+        .then(({body}) => {
+            const {comment} = body;
+            expect(comment).toEqual({
+                author: "lurker",
+                body:"This is just something I want to test, Thank you!"
+            })
+        })
```
Comment (nc-niamh-smith) [inline __tests__/app.test.js:206]: ^^ Is this everything you would expect to be on a comment that exists in the database? Have a think about what else would be on the object - i.e. created_at etc :)

File: __tests__/app.test.js [hunk 2]
```diff
@@ -188,4 +188,87 @@ describe("/api/articles/:article_id/comments", ()=> {
...
+            username:"lurker",
+            body:""
+        })
+        .expect(400)
+        .then(({body}) => {
+            expect(body.msg).toBe("Bad request-The comment is empty")
+        })
+    })
+
+
```
Comment (nc-niamh-smith) [inline __tests__/app.test.js:273]: Excellent test suite and coverage of what the endpoint should and should not accept! I'd like to see you implement the changes I've highlighted above as well :)

File: app.js [hunk 3]
```diff
@@ -14,16 +14,21 @@ app.get("/api/topics", getAllTopics)
...
+
 
 app.get("*",(req,res,next) => {
     res.status(404).send({ msg: "Route does not exist" })
 })
 
 
 app.use((err, req, res, next) => {
-    if (err.code === "22P02") {
+    if ( ["22P02","23502"].includes(err.code) ) {
```
Comment (nc-niamh-smith) [inline app.js:26]: nice use of .includes here!

File: app.js [hunk 4]
```diff
@@ -14,16 +14,21 @@ app.get("/api/topics", getAllTopics)
...
         res.status(400).send({msg: "Bad request!"});
-    } else {
+    } 
+    if(err.code === "23503"){
+        res.status(404).send({msg: "Not found"});
+    }else {
         next(err);
     }
```
Comment (nc-niamh-smith) [inline app.js:33]: I would suggest creating an error handlers file at this point to keep your app from becoming saturated with error handlers :)

File: models/comments_models.js [hunk 5]
```diff
@@ -11,4 +12,16 @@ function fetchCommentsByArticleId(article_id){
...
+    const {username, body} = commentObject
+    if(body === ""){
+        return Promise.reject({status: 400, msg: "Bad request-The comment is empty"})
+    }
+    return db.query(`INSERT into comments (body, article_id,author) VALUES ($1,$2,$3) RETURNING *`, [body,article_id,username])
+    .then(({rows}) => {
+        const {author, body} = rows[0]
+        const comment = {author, body}
+        return comment;
+    })
```
Comment (nc-niamh-smith) [inline models/comments_models.js:25]: ^^ this is where you will make changes because of the test :) Have a look at what is in rows[0]! :)

**Review by nc-niamh-smith — COMMENTED** [review]
Excellent coverage of what is required of the endpoint - I'd like to see more returned on the comment object - as you'll probably want them for your frontend development :)
